language/move-lang/src/shared/mod.rs

```diff
@@ -18,6 +19,33 @@ pub mod remembering_unique_map;
 pub mod unique_map;
 pub mod unique_set;
 
+//**************************************************************************************************
+// Numbers
+//**************************************************************************************************
+
+fn determine_num_text_and_base(s: &str) -> (&str, u32) {
+    match s.strip_prefix("0x") {
+        Some(s_hex) => (s_hex, 16),
+        None => (s, 10),
+    }
+}
+
+// Parse a u8 from a decimal or hex encoding
```

Comment:
Bit strange that you've got a comment only on the u8 version.... Probably doesn't add much